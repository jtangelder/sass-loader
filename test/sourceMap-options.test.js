/**
 * @jest-environment node
 */
import fs from 'fs';
import path from 'path';

import nodeSass from 'node-sass';
import dartSass from 'sass';

import {
  compile,
  getTestId,
  getCodeFromBundle,
  getImplementationByName,
} from './helpers';

const implementations = [nodeSass, dartSass];
const syntaxStyles = ['scss', 'sass'];

describe('sourceMap option', () => {
  implementations.forEach((implementation) => {
    syntaxStyles.forEach((syntax) => {
      const [implementationName] = implementation.info.split('\t');

      it(`not specify (${implementationName}) (${syntax})`, async () => {
        const testId = getTestId('language', syntax);
        const options = {
          implementation: getImplementationByName(implementationName),
          // eslint-disable-next-line no-undefined
          sourceMap: undefined,
        };
        const stats = await compile(testId, { loader: { options } });
        const { css, sourceMap } = getCodeFromBundle(stats);

        expect(css).toBeDefined();
        expect(sourceMap).toBeUndefined();

        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
      });

      it(`false (${implementationName}) (${syntax})`, async () => {
        const testId = getTestId('language', syntax);
        const options = {
          implementation: getImplementationByName(implementationName),
          sourceMap: false,
        };
        const stats = await compile(testId, { loader: { options } });
        const { css, sourceMap } = getCodeFromBundle(stats);

        expect(css).toBeDefined();
        expect(sourceMap).toBeUndefined();

        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
      });

      it(`true (${implementationName}) (${syntax})`, async () => {
        expect.assertions(9);

        const testId = getTestId('language', syntax);
        const options = {
          implementation: getImplementationByName(implementationName),
          sourceMap: true,
        };
        const stats = await compile(testId, { loader: { options } });
        const { css, sourceMap } = getCodeFromBundle(stats);

        expect(css).toBeDefined();
        expect(sourceMap).toBeDefined();

        expect(sourceMap.file).toBeUndefined();
        expect(sourceMap.sourceRoot).toBeDefined();

        expect(sourceMap.sources).toHaveLength(2);

        sourceMap.sources.forEach((sourcePath) => {
          expect(
            fs.existsSync(path.resolve(sourceMap.sourceRoot, sourcePath))
          ).toBe(true);
        });

        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
      });
    });
  });
});
