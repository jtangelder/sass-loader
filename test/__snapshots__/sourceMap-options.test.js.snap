// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sourceMap option false (dart-sass) (sass): errors 1`] = `Array []`;

exports[`sourceMap option false (dart-sass) (sass): warnings 1`] = `Array []`;

exports[`sourceMap option false (dart-sass) (scss): errors 1`] = `Array []`;

exports[`sourceMap option false (dart-sass) (scss): warnings 1`] = `Array []`;

exports[`sourceMap option false (node-sass) (sass): errors 1`] = `Array []`;

exports[`sourceMap option false (node-sass) (sass): warnings 1`] = `Array []`;

exports[`sourceMap option false (node-sass) (scss): errors 1`] = `Array []`;

exports[`sourceMap option false (node-sass) (scss): warnings 1`] = `Array []`;

exports[`sourceMap option not specify (dart-sass) (sass): errors 1`] = `Array []`;

exports[`sourceMap option not specify (dart-sass) (sass): warnings 1`] = `Array []`;

exports[`sourceMap option not specify (dart-sass) (scss): errors 1`] = `Array []`;

exports[`sourceMap option not specify (dart-sass) (scss): warnings 1`] = `Array []`;

exports[`sourceMap option not specify (node-sass) (sass): errors 1`] = `Array []`;

exports[`sourceMap option not specify (node-sass) (sass): warnings 1`] = `Array []`;

exports[`sourceMap option not specify (node-sass) (scss): errors 1`] = `Array []`;

exports[`sourceMap option not specify (node-sass) (scss): warnings 1`] = `Array []`;

exports[`sourceMap option true (dart-sass) (sass): errors 1`] = `Array []`;

exports[`sourceMap option true (dart-sass) (sass): warnings 1`] = `Array []`;

exports[`sourceMap option true (dart-sass) (scss): errors 1`] = `Array []`;

exports[`sourceMap option true (dart-sass) (scss): warnings 1`] = `Array []`;

exports[`sourceMap option true (node-sass) (sass): errors 1`] = `Array []`;

exports[`sourceMap option true (node-sass) (sass): warnings 1`] = `Array []`;

exports[`sourceMap option true (node-sass) (scss): errors 1`] = `Array []`;

exports[`sourceMap option true (node-sass) (scss): warnings 1`] = `Array []`;
